### PR TITLE
[SW-21] Lock history table on migration

### PR DESCRIPTION
**Pull request by AndriyBosik**
[UPGRADE] Logs output on PostgreSQL:
image

[UPGRADE] Logs output on Oracle:
image

[DOWNGRADE] Logs output on PostgreSQL:
image

[DOWNGRADE] Logs output on Oracle:
image

### CODE REVIEW

File: schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java [hunk 1]
```diff
@@ -1,31 +1,56 @@
...
         this.migrationAnalyzer = migrationAnalyzer;
+        this.transactionService = transactionService;
+        this.historyTable = historyTable;
     }
 
     public void up() {
-        var upgradeMigrations = migrationAnalyzer.upgradeAnalyze();
-        migrationRunner.upgrade(upgradeMigrations);
+        doWithinHistoryTable(() -> {
+            List<MigrationData> upgradeMigrations = migrationAnalyzer.upgradeAnalyze();
```
Comment (YuraKovalenko) [inline schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java:33]: why we can't use it like this -  migrationRunner.upgrade(migrationAnalyzer.upgradeAnalyze());

File: schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java [hunk 2]
```diff
@@ -1,31 +1,56 @@
...
+            migrationRunner.upgrade(upgradeMigrations);
+        });
     }
 
     public void down(int version) {
-        var downgradeMigrations = migrationAnalyzer.downgradeAnalyze(new VersionDowngradeStrategyParameters(version));
-        migrationRunner.downgrade(downgradeMigrations);
+        transactionService.doWithinTransaction(() -> {
+            List<MigrationData> downgradeMigrations = migrationAnalyzer.downgradeAnalyze(new VersionDowngradeStrategyParameters(version));
```
Comment (YuraKovalenko) [inline schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java:40]: the same

File: schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java [hunk 3]
```diff
@@ -1,31 +1,56 @@
...
+            migrationRunner.downgrade(downgradeMigrations);
+        });
     }
 
     public void down(String context) {
-        var downgradeMigrations = migrationAnalyzer.downgradeAnalyze(new ContextDowngradeStrategyParameters(context));
-        migrationRunner.downgrade(downgradeMigrations);
+        transactionService.doWithinTransaction(() -> {
+            List<MigrationData> downgradeMigrations = migrationAnalyzer.downgradeAnalyze(new ContextDowngradeStrategyParameters(context));
```
Comment (YuraKovalenko) [inline schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java:47]: the same

**Review by YuraKovalenko — APPROVED** [review]

File: schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java (hunk 1 above)
Comment (AndriyBosik) [reply schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java:33]: Such call chains makes code harder to read and debug as for me

File: schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java (hunk 2 above)
Comment (AndriyBosik) [reply schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java:40]: the same

File: schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java (hunk 3 above)
Comment (AndriyBosik) [reply schema-wizard-core/src/main/java/org/schemawizard/core/starter/SchemaWizard.java:47]: the same

**Pull request merged**